app/concepts/matestack/ui/core/app/app.js

```diff
@@ -10,12 +9,20 @@ const componentDef = {
   },
   computed: Vuex.mapState({
     asyncTemplate: state => state.pageTemplate,
+    currentPathName: state => state.currentPathName,
+    currentSearch: state => state.currentSearch,
+    currentOrigin: state => state.currentOrigin,
```

Comment:
I'm a bit confused that we'd need this. Yes, I read your comment it seems weird to me that there should be no way to access that state throughout the history/event API etc. :thinking: 

And I mean I also tried it out (not with the back button though obviously :sweat_smile: ) and there I had it.

Reply:
Hmkay, seems more involved. Odd, I guess I have to read up on the history APIs etc. https://stackoverflow.com/questions/8980255/how-do-i-retrieve-if-the-popstate-event-comes-from-back-or-forward-actions-with

Reply:
I tried a lot, but this was the only way to fix it.